pyomo/contrib/pyros/separation_problem_methods.py

```diff
@@ -0,0 +1,482 @@
+            else:
+                matrix_of_violations[i][j] = max(solve_data_list[indices_of_violating_realizations[i]][0].list_of_scaled_violations[j], 0)
+
+    sums = []
+    for i in range(matrix_of_violations.shape[1]):
+        sum = 0
+        column = matrix_of_violations[:, i]
+        for j in range(len(column)):
+            sum += column[j]
+        sums.append(sum)
+    max_value = max(sums)
+    idx_i = sums.index(max_value)
+
+    if is_discrete_scenarios:
+        idx_j = violation_dict[idx_i]
+
+    return idx_i, idx_j
+
+
+def solve_separation_problem(model_data, config):
+
+    # Timing variables
+    global_solve_time = 0
+    local_solve_time = 0
+
+    # List of objective functions
+    objectives_map = model_data.separation_model.util.map_obj_to_constr_names
+    constraint_map_to_master = model_data.separation_model.util.map_constr_list_names_to_original_con_names
+
+    # Add additional or remaining separation objectives to the dict
+    # (those either not assigned an explicit priority or those added by Pyros for ssv bounds)
+    sep_priority_dict = config.separation_priority_order
+    for perf_con in model_data.separation_model.util.performance_constraints:
+        if perf_con.name not in sep_priority_dict.keys():
+            sep_priority_dict[perf_con.name] = 0
+
+    # "Bin" the objectives based on priorities
+    sorted_unique_priorities = sorted(list(set(sep_priority_dict.values())), reverse=True)
+
+    for is_global in (False, True):
+        solver = config.global_solver if \
+            (is_global or config.bypass_local_separation) else config.local_solver
+        solve_data_list = []
+
+        for val in sorted_unique_priorities:
+            # Descending ordered by value
+            # The list of performance constraints with this priority
+            perf_constraints = [constr_name for constr_name, priority in sep_priority_dict.items() if priority == val]
+            for perf_con in perf_constraints:
+                #config.progress_logger.info("Separating constraint " + str(perf_con))
+                try:
+                    separation_obj = objectives_map[perf_con]
+                except:
+                    raise ValueError("Error in mapping separation objective to its master constraint form.")
+                separation_obj.activate()
+
+                if perf_con in list(c.local_name for c in model_data.master_nominal_scenario.component_data_objects(Constraint, descend_into=True)):
+                    val = perf_con
+                else:
+                    val = constraint_map_to_master[perf_con]
+
+                if "[" and "]" in val:
+                    # For handling indexed components here, since find_component isn't implemented for IndexedBlock
+                    # want to get the nominal scenario via find_component, not sure if there is a better way to do this
```

Comment:
I see the issue with the logical `and` there. Here, I just need to get the value which the constraint (named `val` here but I am changing that to be clearer) evaluates to in the nominal realization (scenario[0,0]). That value is used to normalize what is considered a violation. I think the below doesn't work in the case that the constraint is indexed since `val` is the name of the constraint.

`getattr(model_data.master_nominal_scenario, val)`

Reply:
I think the correct thing to do here is just the following in all cases:

`model_data.master_nominal_scenario_value = value(model_data.master_nominal_scenario.find_component(val))`